README.md

```diff
@@ -85,6 +88,165 @@ public class TokenGenerationUtil {
+
+[Example](https://github.com/skyflowapi/skyflow-java/blob/master/samples/src/main/java/com/example/ScopedTokenGeneration.java):
+
+```java
+import java.io.File;
+
+public class ScopedTokenGeneration {
+    public static void main(String args[]) {
+
+        String scopedToken = null;
+
+        // Generate Scoped Token  by specifying credentials.json file path
+        try {
+            String filePath = "<YOUR_CREDENTIALS_FILE_PATH>";
+            BearerToken token = new BearerToken.BearerTokenBuilder()
+                    .setCredentials(new File(filePath))
+                    .setRoles(new String[]{"roleID"})
+                    .build();
+
+            scopedToken = token.getBearerToken();
+            System.out.println(scopedToken);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+}
+```
+Note: 
+- Either credentials file path or credentials as string can be passed to `setCredentials` method of BearerTokenBuilder class.
+- If both credentials file path and credentials string are passed to `setCredentials` method, then setter which is at the last will be taken precedence over the other.
+
+## Signed Data Tokens Generation
+
+Skyflow data tokens are generated when sensitive data is inserted into the vault. These data tokens can be digitally signed with the private key of the service account credentials provided, which adds an additional layer of protection. These signed tokens can then be detokenized by passing a signed data token and a bearer token generated from service account credentials having appropriate permissions and context to detokenize API calls.
+
+
+[Example](https://github.com/skyflowapi/skyflow-java/blob/master/samples/src/main/java/com/example/SignedTokenGeneration.java):
+
+``` java
+
+import com.skyflow.errors.SkyflowException;
+import java.io.File;
+import java.util.List;
+
+public class SignedTokenGeneration {
+    public static void main(String args[]) {
+
+        List<SignedDataTokenResponse> signedTokenValue;
+
+        // Generate BearerToken with context by specifying credentials.json file path
+        try {
+            String filePath = "<YOUR_CREDENTIALS_FILE_PATH>";
+            String context = "abc";
+            SignedDataTokens signedToken = new SignedDataTokens.SignedDataTokensBuilder()
+                    .setCredentials(new File(filePath))
+                    .setContext(context)
+                    .setTimeToLive(30.0) // in seconds
+                    .setDataTokens(new String[]{"dataToken1"}).build();
+
+            signedTokenValue =  signedToken.getSignedDataTokens();
+            System.out.println(signedTokenValue);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        // Generate BearerToken with context by specifying credentials.json as string
```

Comment:
Generate signed data tokens using a service account key string, context information, and a time to live.